libcontainer/rootfs_linux.go

```diff
@@ -167,9 +167,18 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 			return err
 		}
 		// bind mount won't change mount options, we need remount to make mount options effective.
-		if err := remount(m, rootfs); err != nil {
+		// first check that we have new options required before attempting a remount
+		buf := syscall.Statfs_t{}
+		if err := syscall.Statfs(dest, &buf); err != nil {
 			return err
 		}
+		if m.Flags&^syscall.MS_REC != (m.Flags&^syscall.MS_REC)&int(buf.Flags) {
```

Comment:
I was thinking that we could check m.Flags itself and if it has just the basic flags for bind mounts then we don't remount.